src/compiler/parser.ts

```diff
@@ -4179,6 +4209,23 @@ namespace ts {
             return finishNode(node);
         }
 
+        function parseJsxClosingFragment(inExpressionContext: boolean): JsxClosingFragment {

```

Comment:
Reading the spec, I don't see how we'd legally get into the `inExpressionContext = false` case since you can't write `<foo><><div /></></foo>`

Reply:
The spec changed, now you can.

Reply:
https://github.com/facebook/jsx/pull/97